ui/src/mixins/lazy-loading-filter.js

```diff
@@ -0,0 +1,178 @@
+
+      const defaultParams = {
+        limit: 48
+      }
+
+      return {
+        url: url || '',
+        params: {
+          ...defaultParams,
+          ...params
+        },
+        decamelizeFieldName: decamelizeFieldName === undefined ? true : decamelizeFieldName
+      }
+    },
+
+    hasFilteredOptions () {
+      return !!this.filteredOptions.length
+    },
+
+    virtualScrollClassName () {
+      return `virtual-scroll-${this.name}`
+    }
+  },
+
+  methods: {
+    async filterOptionsByStore (value) {
+      this.filteredOptions = []
+      this.search = value
+      this.pagination = {
+        page: 1,
+        lastPage: null
+      }
+      this.filteredOptions = await this.fetchOptions()
+    },
+
+    async fetchOptions () {
+      try {
+        if (!this.entity) throw new Error(this.missingPropMessage('entity'))
+        if (!this.name) throw new Error(this.missingPropMessage('name'))
+
+        this.hasFetchError = false
+        this.isLoading = true
+
+        const { url, params, decamelizeFieldName } = this.lazyLoadingPropsWithDefaults
+
+        const { data } = await this.$store.dispatch(`${this.entity}/fetchFieldOptions`, {
+          url,
+          field: decamelizeFieldName ? decamelize(this.name, { separator: '-' }) : this.name,
+          params: {
+            ...params,
+            search: this.search,
+            offset: (this.pagination.page - 1) * params.limit
+          }
+        })
+
+        const { results, count } = data
+
+        this.pagination = {
+          page: this.pagination.page + 1,
+          lastPage: Math.ceil(count / params.limit)
+        }
+
+        this.$emit('fetch-options-success', data)
+
+        return this.handleOptions(results)
+      } catch (error) {
+        console.log(error)
+
+        this.hasFetchError = true
+        this.$emit('fetch-options-error', error)
+
+        return []
+      } finally {
+        this.isLoading = false
+      }
+    },
+
+    async onVirtualScroll ({ index }) {
+      if (this.isScrolling || !this.useLazyLoading) return
+
+      const { scrollContainer, top } = this.getScrollContainerTop()
+      const { lastPage, page } = this.pagination
+      const lastIndex = this.filteredOptions.length - 1
+
+      /**
+       *  if the last page was not reached
+       *  if the scroll container is at the bottom
+       *  if it's not filtering
+       */
+      const canFetchOptions = lastPage && page <= lastPage && index === lastIndex && !this.isLoading
+
+      if (canFetchOptions) {
+        this.isScrolling = true
+
+        const options = await this.fetchOptions()
+        this.filteredOptions.push(...options)
+
+        // solution based on Quasar Select filtering
+        setTimeout(() => {
+          scrollContainer.scrollTo({ top })
+
+          // this is to prevent the virtual-scroll event to be fired again
+          this.$nextTick(() => {
+            this.isScrolling = false
+          })
+        }, 100)
```

Comment:
Aqui eu estava pensando, lembra aquele problema que vai dando vários scroll e ele vai perdendo a precisão? E se o `scrollContainer.scrollTo({ top })` estivesse dentro do `this.$nextTick`, será que ajuda? chegou a testar?

Reply:
Não rolou